### PR TITLE
Gal 473/lsoa midpoint calc

### PR DESCRIPTION
**Pull request by Abdul-NHS**
## Description
https://nhsd-jira.digital.nhs.uk/browse/GAL-473

## Context
Creates the midpoint for each lsoa and uploads to dynamo

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by msandhuNHS — APPROVED** [review]

File: .github/actions/seeding/action.yaml [hunk 1]
```diff
@@ -12,13 +12,11 @@ runs:
     - name: Configure AWS credentials from AWS account
       uses: aws-actions/configure-aws-credentials@v2
       with:
-        role-to-assume: ${{ env.AWS_ROLE_ARN }}
-        aws-region: ${{ env.AWS_REGION }}
+        role-to-assume: arn:aws:iam::136293001324:role/github-oidc-invitations-role
+        aws-region: eu-west-2
```
[the remark this answers is not in the document]
Comment (Abdul-NHS) [reply .github/actions/seeding/action.yaml:16]: its merging into 403 branch and this branch has the old format for terraform code which will be resolved in a future pr for 403

File: scripts/pipeline/nonprod_postcode_load/nonprod_postcode_load.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                            'S': f'{NHS_ENG_REGION}'
+                        },
+                        'SUB_ICB': {
+                            'S': f'{SUB_ICB}'
+                        },
+                        'CANCER_REGISTRY': {
+                            'S': f'{CANCER_REGISTRY}'
+                        },
+                        'EASTING_1M': {
+                            'N': f'{EASTING_1M}'
```
[the remark this answers is not in the document]
Comment (Abdul-NHS) [reply scripts/pipeline/nonprod_postcode_load/nonprod_postcode_load.py:61]: These are fine as they are put in as numbers in dynamo

File: scripts/pipeline/nonprod_postcode_load/nonprod_postcode_load.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                        },
+                        'NORTHING_1M': {
+                            'N': f'{NORTHING_1M}'
```
[the remark this answers is not in the document]
Comment (Abdul-NHS) [reply scripts/pipeline/nonprod_postcode_load/nonprod_postcode_load.py:64]: These are fine as they are put in as numbers in dynamo

File: scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  fs.writeFile(filename, obj, (err) => {
+    if (err) {
+      console.log(err);
+    }
+    console.log("CSV file saved successfully");
+  });
+};
+
+const lsoaHeader =
+  "POSTCODE,POSTCODE_2,LOCAL_AUT_ORG,NHS_ENG_REGION,SUB_ICB,CANCER_REGISTRY,EASTING_1M,NORTHING_1M,LSOA_2011,MSOA_2011,CANCER_ALLIANCE,ICB,OA_2021,LSOA_2021,MSOA_2021,IMD_RANK,IMD_DECILE,AVG_EASTING,AVG_NORTHING";
```
[the remark this answers is not in the document]
Comment (Abdul-NHS) [reply scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.js]: added concat string

**Review by msandhuNHS — APPROVED** [review]

**Review by zainm97 — APPROVED** [review]
commenting out code will be addressed in 403

**Pull request merged**